src/crimson/osd/ops_executer.h

```diff
@@ -529,7 +525,9 @@ OpsExecuter::flush_changes_n_do_ops_effects(
     return maybe_mutated;
   } else {
     return maybe_mutated.then_unpack_interruptible(
-      [this, pg=std::move(pg)](auto&& submitted, auto&& all_completed) mutable {
+      // need extra ref pg due to apply_stats() which can be executed after
+      // informing snap mapper
+      [this, pg=this->pg](auto&& submitted, auto&& all_completed) mutable {
```

Comment:
After fixing this problem the snaptrimmer branch fails the `workunits/rados/test.sh` exactly like a fresh `main`:

```
[rzarzynski@o06 build]$ grep -A10 terminate out/osd.1.stdout 
terminate called after throwing an instance of 'std::runtime_error'
  what():  read gave enoent on #-1:a95ba935:::osdmap.579:0#
Aborting on shard 0.
Backtrace:
 0# gsignal in /lib64/libc.so.6
 1# abort in /lib64/libc.so.6
 2# 0x00007F16BDA2709B in /lib64/libstdc++.so.6
 3# 0x00007F16BDA2D53C in /lib64/libstdc++.so.6
 4# 0x00007F16BDA2C559 in /lib64/libstdc++.so.6
 5# __gxx_personality_v0 in /lib64/libstdc++.so.6
 6# 0x00007F16BC6EBB03 in /lib64/libgcc_s.so.1
[rzarzynski@o06 build]$ git describe
v18.0.0-2140-gfa8a9c73ae3
[rzarzynski@o06 build]$ git log | head -n 4
commit fa8a9c73ae3a5cc905789e96e76c4f9d3a0b0573
Merge: 8b5ab2bc038 9d1fb1bc39a
Author: zdover23 <zac.dover@proton.me>
Date:   Fri Feb 3 04:20:48 2023 +1000
```